agent/command/macsign.go

```diff
@@ -204,7 +205,12 @@ func (macSign *macSign) Execute(ctx context.Context,
 		args = append(args, "--verify")
 	}
 
+	envs := []string{
+		fmt.Sprintf("MACOS_NOTARY_SECRET=%s", macSign.Secret),
+	}
+
 	cmd := exec.Command(macSign.ClientBinary, args...)
+	cmd.Env = append(os.Environ(), envs...)
```

Comment:
I'm not sure quite honestly! I only added that to match the current behavior, since we weren't setting the Env field and it will default to `os.Environ()` anyway.

Reply:
Hmm in that case, I'm inclined to pass just the one env var if we can.

Reply:
After looking through I suspect that at least `PATH` is being implicitly depended on by the Macnotary client for a couple of things. If we change the behavior to no longer send everything down, we'd probably need to make sure we run some actual tests so we don't break things.

Reply:
That makes sense. In that case, I'm inclined to say this should use the minimum set of env vars needed to function + test it (at least in staging). Passing all the env vars down by default to an arbitrary process is a big enough security concern to warrant trimming it down.

Reply:
While I agree that we need to improve security across the board, I want to point out that this change doesn't negatively impact security in Evergreen. And requested changes to this PR don't improve the security either.
Any user in Evergreen can easily use shell.exec in their task to print all the env vars or pass them to an arbitrary process and we don't have anything in place to prevent that.
Also, mas signer is not really an arbitrary process. It's controlled by one of the internal teams. We know exactly what it does.
And in this particular case we're not even going to pass more variables to the process than we used to.

[EVG-20875](https://jira.mongodb.org/browse/EVG-20875) has been downgraded and we've been leaking a lot of secrets due to that behavior this whole time. We wouldn't even need this change if we stopped collecting all the ps output.

Reply:
Hmm for this one, it's not really possible to write a unit test in Evergreen itself since it's really just wrapping args to pass to the mac client.

Since it's not really possible to test in staging due to the lack of mac host, I imagine you can try writing a small script on your machine (assuming it's a Mac) that invokes the binary. Alternatively, you can just test this in prod by adding a temporary (undocumented) parameter to mac.sign that would let you pass in the env vars, which you can manually pass in and see what env vars it actually needs to function.

Reply:
@zakhark The part about shell/subprocess.exec being insecure due to lack of redaction is true, but the part about the mac sign helper being more secure is not true because this command allows you to pass [an arbitrary binary](https://github.com/evergreen-ci/evergreen/blob/420fdba876b90578598d99c6dce597de3b1657b8/agent/command/macsign.go#L150) - if we were really trying to be malicious, either swapping out `/usr/local/bin/macnotary` for a homemade binary or just passing an alternative binary into this command is pretty much the same attack vector as `shell.exec`/`subprocess.exec` offers with `env`.

I don't have particular opinions about this command, since it's really just a subprocess.exec wrapper designed for one specific binary, so I'm fine if we want to defer this to the redaction project.

Reply:
@Kimchelly That's good information! Zakhar and I chatted and I think we're OK pausing this for now until Tural gets back and we can make a decision about whether to keep this around and fix it properly or just remove mac.sign if no one is using it in favor for just running the binary themselves.